testsuite/smoke-test/src/randomness/long_running_crash_recovery.rs

```diff
@@ -0,0 +1,174 @@
+    let i = started_nodes[thread_rng().gen_range(0, started_nodes.len())];
+    println!("Action: stop node {}", i);
+    println!();
+    swarm.validators_mut().nth(i).unwrap().stop();
+    validator_power_status_vec[i] = false;
+}
+
+async fn set_random_vtxn_and_randomness_config(rest_cli: &Client, aptos_cli: &CliTestFramework, root_idx: usize) {
+    let target_vtxn_status = thread_rng().gen_range(0, 2);
+    let target_randomness_status = thread_rng().gen_range(0, 3);
+    println!("Action: set target_vtxn_status={}, target_randomness_status={}", target_vtxn_status, target_randomness_status);
+    println!();
+
+    let mut consensus_config = get_current_consensus_config(rest_cli).await;
+    if target_vtxn_status == 0 {
+        consensus_config.disable_validator_txns();
+    } else {
+        consensus_config.enable_validator_txns();
+    }
+    let consensus_config_bytes = bcs::to_bytes(&consensus_config).unwrap();
+    let script = format!(r#"
+script {{
+    use aptos_framework::aptos_governance;
+    use aptos_framework::consensus_config;
+    use aptos_framework::randomness_config;
+    use aptos_std::fixed_point64;
+
+    fun main(core_resources: &signer) {{
+        let framework_signer = aptos_governance::get_signer_testnet_only(core_resources, @0x1);
+        let consensus_config_bytes = vector{:?};
+        consensus_config::set_for_next_epoch(&framework_signer, consensus_config_bytes);
+        let target_randomness_status = {};
+        let randomness_config = if (target_randomness_status == 0) {{
+            randomness_config::new_off()
+        }} else if (target_randomness_status == 1) {{
+            randomness_config::new_v1(
+                fixed_point64::create_from_rational(1, 2),
+                fixed_point64::create_from_rational(2, 3)
+            )
+        }} else {{
+            randomness_config::new_v2(
+                fixed_point64::create_from_rational(1, 2),
+                fixed_point64::create_from_rational(2, 3),
+                fixed_point64::create_from_rational(2, 3),
+            )
+        }};
+        randomness_config::set_for_next_epoch(&framework_signer, randomness_config);
+        aptos_governance::reconfigure(&framework_signer);
+    }}
+}}
+"#, consensus_config_bytes, target_randomness_status);
+    let txn_result = aptos_cli.run_script(root_idx, script.as_str()).await.unwrap();
+    assert!(txn_result.success.unwrap());
+}
+
+async fn make_change(validator_power_status_vec: &mut Vec<bool>, rest_cli: &Client, aptos_cli: &CliTestFramework, root_idx: usize, swarm: &mut LocalSwarm) {
+    let stopped_nodes: Vec<usize> = validator_power_status_vec.iter().enumerate().filter_map(|(idx, &power_on)|if !power_on {Some(idx)} else {None}).collect();
+    match stopped_nodes.len() {
+        0 => {
+            match thread_rng().gen_range(0.0, 1.0) {
```

Comment:
Why float not integer for random number?

Reply:
not too much difference?